### ISSUE TITLE
🐛  poetry parsing invalid when dependency blocks are not indented 

### ISSUE THREAD

**Issue by lachaib** [Twixes/pypi-assistant#33]
Hello,

Maybe it's because my project was initiated with a very old version of `poetry init`, but the `[tool.poetry.dependencies]` and `[tool.poetry.group.dev.dependencies]` weren't indented, and as a result this extension wasn't able to properly parse it.

Expected output:
```
Parsed requirements in file:///[...]/pyproject.toml, v2:
python @ 30#4 - 30#20
timezonefinder @ 31#4 - 31#29
alembic @ 32#4 - 32#23
beautifulsoup4 @ 33#4 - 33#30
blinker @ 34#4 - 34#22
boto3 @ 35#4 - 35#22
countryinfo @ 36#4 - 36#25
```

Actual output:
```
Parsed requirements in file:///Users/louisamaury.chaib/projects/partoo/pyproject.toml, v1:
jobs @ 181#0 - 181#10
max-line-length @ 185#0 - 185#23
show-fixes @ 194#0 - 194#17
line-length @ 196#0 - 196#17
```
These above settings are pylint & ruff configuration which were properly indented.

Indenting the dependencies blocks work just fine with no leftovers, so I'm assuming is just a matter of detecting that the toml stack was part a child block `[tool.poetry]`, however it seems strange that it didn't properly detect exiting the poetry block.

So to summarize:
* need of support of unindented poetry dependencies blocks
* need to make sure other blocks aren't read by mistake

Thanks a ton

**Comment by Twixes** [comment]
Hi @lachaib, thanks for the report! Can you please attach the `pyproject.toml` file that isn't processed correctly? This will allow me to reproduce the issue and take a stab at it

**Comment by lachaib** [comment]
Hello,
Fun fact, after reloading VS Code this morning to open the file, now it's properly parsed 🙄 
I genuinely can't tell why it was working before and why it isn't now.

I'll close this issue, apologies
